### PR TITLE
Return documents in order of ids that were provided

### PR DESCRIPTION
**Pull request by jameshawkins**
Related to https://github.com/OrchardCMS/OrchardCore/issues/2651

### CODE REVIEW

File: test/YesSql.Tests/CoreTests.cs [hunk 1]
```diff
@@ -2301,6 +2301,39 @@ public async Task ShouldGetDynamicById()
...
+                    session.Save(circle);
+                    circleIds.Add(circle.Id);
+                }
+
+                await session.CommitAsync();
+            }
+
+            using (var session = _store.CreateSession())
+            {
+                circleIds.Reverse();
```
Comment (sebastienros) [inline test/YesSql.Tests/CoreTests.cs]: The test is wrong, I don't think `Reverse()` mutates the callee. It returns a new collection instead. So your test is still testing the same initial order.

```suggestion
                circleIds = circleIds.Reverse();
```

Comment (sebastienros) [reply test/YesSql.Tests/CoreTests.cs]: Add `.ToList()` to that

Comment (jameshawkins) [reply test/YesSql.Tests/CoreTests.cs]: it mutates the list: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.reverse?view=netframework-4.7.2

Comment (jameshawkins) [reply test/YesSql.Tests/CoreTests.cs]: i was surprised too :)

**Comment by sebastienros** [comment]
Now just wait for the build to finish and update the version on Orchard in a PR

**Comment by jameshawkins** [comment]
Orchard PR submitted: https://github.com/OrchardCMS/OrchardCore/pull/2657

**Pull request merged**